cmd/rook/rook/rook.go

```diff
@@ -217,30 +217,54 @@ func GetOperatorBaseImageCephVersion(context *clusterd.Context) (string, error)
 // GetInternalOrExternalClient will get a Kubernetes client interface from the KUBECONFIG variable
 // if it is set, or from the operator pod environment otherwise.
 func GetInternalOrExternalClient() kubernetes.Interface {
-	kubeconfig := os.Getenv("KUBECONFIG")
-
-	var (
-		config *rest.Config
-		err    error = fmt.Errorf("did not attempt to load a client") // just in case
-	)
+	var restConfig *rest.Config
+	var clientset *kubernetes.Clientset
+	var err error
 
+	// if KUBECONFIG env var is present, it is the source of truth
+	kubeconfig := os.Getenv("KUBECONFIG")
 	if kubeconfig != "" {
+		logger.Debugf("attempting to create kube client interface from KUBCONFIG environment variable: %s", kubeconfig)
 		for _, kConf := range strings.Split(kubeconfig, ":") {
-			config, err = clientcmd.BuildConfigFromFlags("", kConf)
+			restConfig, err = clientcmd.BuildConfigFromFlags("", kConf)
 			if err == nil {
-				break
+				logger.Debugf("attmepting to create kube clientset from kube config file %q", kConf)
+				clientset, err = kubernetes.NewForConfig(restConfig)
+				if err == nil {
+					logger.Infof("created kube client interface from kube config file %q present in KUBECONFIG environment variable", kConf)
+					return clientset
+				}
 			}
 		}
-	} else {
-		config, err = rest.InClusterConfig()
 	}
 	if err != nil {
-		TerminateOnError(err, "error creating client")
+		TerminateOnError(err, "could not create kube client interface from KUBECONFIG environment variable: "+kubeconfig+"; "+
+			"if the KUBECONFIG environment variable is incorrect, unset it, or set it to the correct kube config file location")
 	}
 
-	client, err := kubernetes.NewForConfig(config)
+	logger.Debug("attempting to create kube client interface using default CLI parameters")
+	defaultConfigFlags := genericclioptions.NewConfigFlags(false)
+	restConfig, err = defaultConfigFlags.ToRESTConfig()
+	if err == nil {
+		clientset, err = kubernetes.NewForConfig(restConfig)
+		if err == nil {
+			logger.Infof("created kube client interface from default CLI parameters")
+			return clientset
+		}
+		logger.Debugf("failed to create kube client interface from default CLI parameters; " +
+			"check the default kube config file location to ensure the proper config is present there")
+	}
+
+	logger.Infof("attempting to create kube client interface assuming the application is running in a Kubernetes Pod; " +
```

Comment:
This is the common case for running the operator, right? If so, how about debug level?
```suggestion
	logger.Debug("attempting to create kube client interface assuming the application is running in a Kubernetes Pod; " +
```

Reply:
This is the uncommon case if someone installs using CSV --> they must run from the CLI, and the longer text below is critical to fix problems that have been reported by users.

Reply:
Ah, this is not the path for running in the pod? Makes sense then.